packages/styled-system/src/types.ts

```
@@ -0,0 +1,229 @@
+// First-class TypeScript types — the public surface that replaces @types/styled-system.
+// Foundation types (Theme, ResponsiveValue, ThemeValue, ObjectOrArray) and the
+// generic, theme-scale-aware prop interfaces mirror @types/styled-system@5.1.25 so this
+// package is a drop-in replacement. The one deviation: ThemeValue's inference helper
+// uses `unknown` instead of the original's `any`.
+import type * as CSS from 'csstype'
+
+export type ObjectOrArray<T, K extends keyof never = keyof never> =
+  | T[]
+  | Record<K, T | Record<K, T> | T[]>
```

ESLint

[error] Replace `|·T[]⏎·` with `T[]`

(prettier/prettier)